python/Ganga/Core/GangaRepository/SubJobXMLList.py

```diff
@@ -517,3 +561,52 @@ def _private_display(self, reg_slice, this_format, default_width, markup):
             ds += markup(this_format % tuple(vals), colour)
 
         return ds
+
+    @staticmethod
+    def jobHasChildrenTest(jobDirectory, datafileName):
```

Comment:
I can change the name of the function to ```checkSubJobXML``` and then have a direct call to determine the number of  children in the XML reader.
The whole point of SubJobXMLList class is to care about how the subjobs are stored/accessed/found when on disk. You can't sensibly ask a parent job without starting to reproduce functionality which belongs here. We used to do it elsewhere but then we had no lazy loading for subjobs at all.